src/xdist/workermanage.py

```diff
@@ -261,7 +262,8 @@ def setup(self):
         remote_module = self.config.hook.pytest_xdist_getremotemodule()
         self.channel = self.gateway.remote_exec(remote_module)
         # change sys.path only for remote workers
-        change_sys_path = not self.gateway.spec.popen
+        # restore sys.path from a frozen copy for local workers
+        change_sys_path = _sys_path if self.gateway.spec.popen else None
```

Comment:
we have to use a frozen copy of sys.path because `--import-mode=prepend` changes the sys.path